qcfractal/interface/collections/torsiondrive_dataset.py

```diff
@@ -80,6 +80,7 @@ def add_specification(self,
         spec = TorsionDriveSpecification(
             name=lname, optimization_spec=optimization_spec, qc_spec=qc_spec, description=description)
         self.data.td_specs[lname] = spec
+        self.save()
```

Comment:
Why does the TDDaset's `add_` method here call a `save` back to the Fractal server? This seems to be different behavior from the other `Collection` classes which don't invoke `save` until its manually called or `compute` is called?

Reply:
Yes, Procedure Datasets track their data by ObjectId as our querying is less exact for procedures. Result Datasets only need the molecule ObjectId and it can find the exact result it needs.